discovery-provider/alembic/versions/ed974e76d415_fix_indexing_track_slug.py

```diff
@@ -0,0 +1,414 @@
+"""fix-indexing-track-slug
+
+Revision ID: ed974e76d415
+Revises: cb9aa46f1e46
+Create Date: 2021-07-30 15:24:52.899198
+
+"""
+from alembic import op
+from src.models import Block, Track, TrackRoute, SkippedTransaction
+from sqlalchemy import orm
+
+
+# revision identifiers, used by Alembic.
+revision = "ed974e76d415"
+down_revision = "cb9aa46f1e46"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+
+    # Create the blocks if non-existent
+    blocks = [
+        {
+            "blocknumber": 22477107,
+            "txhash": "0xd16cd27f8635e84cbb6d02ecf6a5304bae998381cda1291d4cb690d9159196d9",
+            "blockhash": "0x82ba068410ac71a70dfe95708b3c4ec08884355d8384423deb2337a6bcd64dd0",
+        },
+        {
+            "blocknumber": 22477427,
+            "txhash": "0x135c236ebb4d02a1d2bc13bf946e982871ad59a29ed413bbe6aed2acb60a1f17",
+            "blockhash": "0x63b16e801cb90f3497043a904e0180a8161d9bbb45add0a9380e277b0befbbba",
+        },
+        {
+            "blocknumber": 22477062,
+            "txhash": "0x05771964bc0527268c330b78bb1819998d589d31a95e5f2f8aa846041de1dfee",
+            "blockhash": "0x8e5f3edd55d1c7f3d6484a8fb7d2a9d825fb53ecb446c8959dd3887fa999d2e7",
+        },
+        {
+            "blocknumber": 22477832,
+            "txhash": "0xda0b64477469390eaa3b17098c01f3e03e5b0e52959ffd493804538dda039a92",
+            "blockhash": "0x705d01ed50cf8f1d83686acc19c9d0bdd52493b3552532543629cdf07fba6005",
+        },
+    ]
+    for block in blocks:
+        block_query = (
+            session.query(Block)
+            .filter(
+                Block.blockhash == block["blockhash"],
+                Block.number == block["blocknumber"],
+            )
+            .first()
+        )
+        if not block_query:
+            session.add(
+                Block(
+                    blockhash=block["blockhash"],
+                    number=block["blocknumber"],
+                    is_current=False,
+                )
+            )
```

Comment:
This should only happen on staging/local dev - is there any impact to having random block rows?

Reply:
what if we did the inverse? if it doesn't exist return out of the migration? all four of those rows are in the blocks table on prod and we only want to run this in the case it exists already

Reply:
^ good idea